server/planning/feed_parsers/superdesk_event_json.py

```diff
@@ -0,0 +1,171 @@
+import logging
+import datetime
+
+from superdesk.io.feed_parsers import FileFeedParser
+import pytz
+import json
+
+utc = pytz.UTC
+logger = logging.getLogger(__name__)
+
+
+class SuperdeskFeedParser(FileFeedParser):
+    """Superdesk event specific parser.
+
+    Feed Parser which can parse the Superdesk Event feed and convert to internal event format,
+    but the firstcreated and versioncreated times are localised.
+    """
+
+    NAME = 'SuperdeskEventJson'
+
+    label = 'Superdesk Event Json'
+
+    def can_parse(self, file_path):
+        try:
+            with open(file_path, 'r') as f:
+                superdesk_event = json.load(f)
+                if superdesk_event.get('guid'):
+                    return True
+        except Exception:
+            pass
+        return False
+
+    def parse(self, file_path, provider=None):
+        self.items = []
+        with open(file_path, 'r') as f:
+            superdesk_event = json.load(f)
+        self.items.append(self._transform_from_superdesk_event(superdesk_event))
+        return self.items
+
+    def _transform_from_superdesk_event(self, superdesk_event):
+        item = {}
+        if superdesk_event.get('_current_version'):
```

Comment:
Most of the fields can be directly copied, as these are generated from Superdesk as well.

It would make it easier to maintain to focus solely on the fields that need special attention.

**Fields to Ignore:**
```python
ignore_fields = [
    'files',  # This could be added at a later date
    'state_reason',
    'schedule_settings',
    'pubstatus',
    '_current_version',  # Versions are not enabled for Events
    '_id',
    'item_id',
    'actioned_date'
]
```

**Fields to manually assign:**
```python
item['_created'] = utcnow()
item['_updated'] = utcnow()
item['state'] = WORKFLOW_STATE.INGESTED
```

**Fields to assign from local CV**
These fields must use items found in the local vocabularies DB, searching via **qcode** of each item, and ignoring those not found.

```python
assign_from_local_cv = [
    'anpa_category',
    'subject',
    'calendars',
    'place',
    'occur_status'  # This must always have a value
]
```

**Fields to lookup entries for**
The following fields contain values that may not exist in the local DB. If the value is not found in the DB, one is to be added and the new **id** assigned when copying these fields
```python
add_to_local_db = [
    'event_contact_info',  # get_resource_service('contacts')
    'location'  # get_resource_service('locations')
]
```

**All other fields**
All other fields can simply be copied across